metals/src/main/scala/scala/meta/internal/troubleshoot/ScalaProblem.scala

```diff
@@ -0,0 +1,75 @@
+package scala.meta.internal.troubleshoot
+
+import scala.meta.internal.metals.BuildInfo
+import scala.meta.internal.metals.ScalaVersions
+
+sealed abstract class ScalaProblem {
+  def message: String
+
+  protected val hint = "run 'Build import' to enable code navigation."
+
+}
+
+// scala version issues
+case class UnsupportedScalaVersion(version: String) extends ScalaProblem {
+  override def message: String = {
+    if (ScalaVersions.isSupportedScalaBinaryVersion(version)) {
+      val recommended = ScalaVersions.recommendedVersion(version)
+      s"Upgrade to Scala $recommended and " + hint
+    } else {
+      val versionToUpgradeTo =
+        if (ScalaVersions.isScala3Version(version)) {
+          s"Scala ${BuildInfo.scala3}"
+        } else {
+          s"Scala ${BuildInfo.scala213} or ${BuildInfo.scala212}"
+        }
+      s"Code navigation is not supported for this compiler version, change to " + versionToUpgradeTo + " and " + hint
+    }
+  }
+}
+
+case class DeprecatedScalaVersion(version: String) extends ScalaProblem {
+  override def message: String =
+    s"Scala $version might not be supported in upcoming versions of Metals, " +
+      s"please upgrade to Scala ${ScalaVersions.recommendedVersion(version)}."
+}
+
+case class FutureScalaVersion(version: String) extends ScalaProblem {
+  override def message: String = s"Scala $version is not yet supported"
+}
+
+// semanticdb problems
+case class SemanticDBDisabled(
+    scalaVersion: String,
+    bloopVersion: String,
+    isBloopVersionSupported: Boolean
+) extends ScalaProblem {
+  override def message: String = {
+    if (isBloopVersionSupported) {
```

Comment:
This confuses me a bit? I'd expect that `isBloopVersionSupported` is true, then we'd have no issue. But this message is saying that it's not supported and Metals requires at least Bloop x. Is this correct?

Reply:
Och yeah, that was actually badly named. Renamed to `unsupportedBloopVersion`

Reply:
haha I'm glad I caught this then.